synapse/storage/events.py

```diff
@@ -2109,10 +2119,20 @@ def _purge_history_txn(
         #
         # So, let's stick it at the end so that we don't block event
         # persistence.
-        logger.info("[purge] updating room_depth")
+        txn.execute("""
+            SELECT COALESCE(MIN(depth), 0)
+            FROM event_backward_extremities AS eb
+            INNER JOIN event_edges AS eg ON eg.prev_event_id = eb.event_id
+            INNER JOIN events AS e ON e.event_id = eg.event_id
```

Comment:
is this looking for the prev_event before the extremity? which we have likely just purged?

Reply:
This is looking at the events that point to the extremity, as we don't actually have the extremity events.

Reply:
(And I think the check is the right way round?)

Reply:
oh right. ehm. I'm not convinced this is going to calculate the right thing in the presence of disconnected bits of dag.

Reply:
Each disconnected part of the DAG will have their backwards extremities in the `event_backward_extremities` table, so the event with the minimum depth must be one of them, no?